test/MetadataUtility.Tests/Metadata/FrontierLabs/FlacHeaderExtractorTests.cs

```diff
@@ -48,13 +46,20 @@ public async void CanProcessFilesWorks(FixtureModel model)
         public async void ProcessFilesWorks(FixtureModel model)
         {
             // we can process all files that have a filename
-            var recording = await this.subject.ProcessFileAsync(
-                model.ToTargetInformation(this.RealFileSystem),
-                this.Recording);
+            var recording = new Recording();
 
-            recording.DurationSeconds?.TotalSeconds.Should().Be(model.DurationSeconds);
+            if (await this.subject.CanProcessAsync(model.ToTargetInformation(this.RealFileSystem)))
```

Comment:
My thinking here was, some of the fixture models (namely the wav header one) isn't going to pass CanProcessAsync, so we don't want it running ProcessAsync. I've changed the if statement to `if (model.IsFlac)`, and used the same if statement in my FlacTests file as well. Let me know your thoughts.

Reply:
I should also note that I changed the MimeType value for the PreallocatedHeader fixtures csv to audio/wav instead of audio/flac, so that it returns false for model.IsFlac.

Reply:
So yeah, these cases will be a little tricky.

If some of our fixtures are not appropriate for some tests, we should filter them out. That is what the `IsFlac` property was for and you correctly keyed into that. It might not be the best property to filter on though.

Then we have to decide what happens for faulty files. For CanProcessAsync the goal is to never have it run if there is file it can't process. That won't always be true, so we'll code defensively anyway, 

On reflection I think your original approach is pretty decent, but we'll tweak it a little bit. Something like:

```
if (!  if (await this.subject.CanProcessAsync(model.ToTargetInformation(this.RealFileSystem)))) {
   Assert.Ignore("Test filtered")
}
```

Reply:
Another approach might be to keep a list of which extractors we expect any fixture to run on. That could be the best approach in the long run.

So in the fixtures CSV we'd have a column for each fixture named `CanProccess`  and `Process`. The values for those columns will be a semi-colon delimited list of which extractors we expect to be tested for each file.

It might be more work in the short run, but it the long run it will allow us to be very explicit about what we test.